cmd/vulcan-nuclei/main.go

```diff
@@ -0,0 +1,362 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os/exec"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	report "github.com/adevinta/vulcan-report"
+
+	check "github.com/adevinta/vulcan-check-sdk"
+	"github.com/adevinta/vulcan-check-sdk/helpers"
+	checkstate "github.com/adevinta/vulcan-check-sdk/state"
+	"github.com/avast/retry-go"
+)
+
+var (
+	checkName              = "vulcan-nuclei"
+	logger                 = check.NewCheckLog(checkName)
+	userAgent              = "x-vulcan-nuclei"
+	nucleiCmd              = "./nuclei"
+	nucleiTemplatePath     = "/root/nuclei-templates/"
+	nucleiTemplateListJSON = "TEMPLATES-STATS.json"
+
+	defaultTemplateExclusionList = []string{
+		"fuzzing",
+		"helpers",
+		"miscellaneous",
+		"workflows",
+	}
+)
+
+type void struct{}
+
+type options struct {
+	SkipUpdateTemplates   bool     `json:"skip_update_templates"`
+	Severities            []string `json:"severities"`
+	TemplateInclusionList []string `json:"template_inclusion_list"`
+	TemplateExclusionList []string `json:"template_exclusion_list"`
+}
+
+// TemplateList defines nuclei templates available.
+type TemplateList struct {
+	Directory []struct {
+		Name  string `json:"name"`
+		Count int    `json:"count"`
+	} `json:"directory"`
+}
+
+type finding struct {
+	name     string
+	severity string
+	template string
+	match    string
+}
+
+func main() {
+	c := check.NewCheckFromHandler(checkName, run)
+	c.RunAndServe()
+}
+
+func runNucleiCmd(args []string) ([]byte, error) {
+	var err error
+	var cmdOutput []byte
+	err = retry.Do(
```

Comment:
Question: why are we doing retries?

Reply:
Well, I had some problems in the past trying to download the templates due to network issues in the local development.
I thought in making in the execution (of any command) more resilient in case of errors.
If you foresee any potential issue with this let me know and I can remove the retry behaviour or make it work only for the template update command.